exporters/otlp/test/recordable_test.cc

```diff
@@ -68,6 +66,122 @@ TEST(Recordable, SetDuration)
 
   EXPECT_EQ(rec.span().end_time_unix_nano(), unix_end);
 }
+
+TEST(Recordable, SetStatus)
+{
+  Recordable rec;
+  trace::CanonicalCode code(trace::CanonicalCode::OK);
+  nostd::string_view description = "For test";
+  rec.SetStatus(code, description);
+
+  EXPECT_EQ(rec.span().status().code(), opentelemetry::proto::trace::v1::Status_StatusCode(code));
+  EXPECT_EQ(rec.span().status().message(), description);
+}
+
+TEST(Recordable, AddEvents)
+{
+  Recordable rec;
+  nostd::string_view name = "Test Event";
+
+  std::chrono::system_clock::time_point event_time = std::chrono::system_clock::now();
+  core::SystemTimestamp event_timestamp(event_time);
+
+  rec.AddEvent(name, event_timestamp);
+
+  uint64_t unix_event_time =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(event_time.time_since_epoch()).count();
+
+  EXPECT_EQ(rec.span().events(0).name(), name);
+  EXPECT_EQ(rec.span().events(0).time_unix_nano(), unix_event_time);
+}
+
+// Test non-int single types. Int single types are tested using templates (see IntAttributeTest)
+TEST(Recordable, SetSingleAtrribute)
+{
+  Recordable rec;
+  common::AttributeValue bool_val(true);
+  rec.SetAttribute("bool_attr", bool_val);
+
+  common::AttributeValue double_val(3.3);
+  rec.SetAttribute("double_attr", double_val);
+
+  common::AttributeValue str_val(nostd::string_view("Test"));
+  rec.SetAttribute("str_attr", str_val);
+
+  EXPECT_EQ(rec.span().attributes(0).value().bool_value(), nostd::get<bool>(bool_val));
+  EXPECT_EQ(rec.span().attributes(1).value().double_value(), nostd::get<double>(double_val));
+  EXPECT_EQ(rec.span().attributes(2).value().string_value(),
+            nostd::get<nostd::string_view>(str_val).data());
+}
+
+// Test non-int array types. Int array types are tested using templates (see IntAttributeTest)
+TEST(Recordable, SetArrayAtrribute)
+{
+  Recordable rec;
+  const int kArraySize = 3;
+
+  bool bool_arr[kArraySize] = {true, false, true};
+  nostd::span<const bool> bool_span(bool_arr);
+  rec.SetAttribute("bool_arr_attr", common::AttributeValue(bool_span));
```

Comment:
You should be able to leave the explicit `common::AttributeValue` out here (and for all following occurences).